hudi-common/src/main/java/org/apache/hudi/internal/schema/InternalSchema.java

```diff
@@ -51,38 +50,34 @@ public class InternalSchema implements Serializable {
   private transient Map<Integer, String> idToName = null;
 
   public static InternalSchema getEmptyInternalSchema() {
-    return new InternalSchema(-1L, new ArrayList<>());
+    return new InternalSchema(-1L, RecordType.get());
   }
 
   public boolean isEmptySchema() {
     return versionId < 0;
   }
 
-  public InternalSchema(List<Field> columns) {
-    this(DEFAULT_VERSION_ID, columns);
+  public InternalSchema(RecordType recordType) {
```

Comment:
These changes are necessary to allow to preserve Avro record's name during conversion to `InternalSchema` and back.

Changes
 - Rebasing ctors to accept `RecordType` instead of list of fields
 - Inlining redundant ctors